clients/fossid-webapp/src/test/kotlin/FossIdClientTestNewProject.kt

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies
+import com.fasterxml.jackson.databind.json.JsonMapper
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldBeEmpty
+import io.kotest.matchers.maps.shouldContain
+import io.kotest.matchers.nulls.shouldBeNull
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+
+import okhttp3.OkHttpClient
+
+import org.ossreviewtoolkit.fossid.FossIdRestService
+import org.ossreviewtoolkit.fossid.api.Scan
+import org.ossreviewtoolkit.fossid.api.identification.ignored.IgnoredFile
+import org.ossreviewtoolkit.fossid.api.identification.markedAsIdentified.MarkedAsIdentifiedFile
+import org.ossreviewtoolkit.fossid.api.status.DownloadStatus
+import org.ossreviewtoolkit.fossid.api.status.ScanState
+import org.ossreviewtoolkit.fossid.checkDownloadStatus
+import org.ossreviewtoolkit.fossid.checkResponse
+import org.ossreviewtoolkit.fossid.checkScanStatus
+import org.ossreviewtoolkit.fossid.createProject
+import org.ossreviewtoolkit.fossid.createScan
+import org.ossreviewtoolkit.fossid.downloadFromGit
+import org.ossreviewtoolkit.fossid.getProject
+import org.ossreviewtoolkit.fossid.listIdentifiedFiles
+import org.ossreviewtoolkit.fossid.listIgnoredFiles
+import org.ossreviewtoolkit.fossid.listMarkedAsIdentifiedFiles
+import org.ossreviewtoolkit.fossid.listScanResults
+import org.ossreviewtoolkit.fossid.listScansForProject
+import org.ossreviewtoolkit.fossid.runScan
+import org.ossreviewtoolkit.fossid.toList
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+
+/**
+ * This client test creates a new project, triggers the download and the scan and gets the scan results
+ */
+class FossIdClientTestNewProject : StringSpec({
+    lateinit var wiremock: WireMockServer
+    lateinit var service: FossIdRestService
```

Comment:
Please avoid `lateinit` where not strictly needed, similar to like I've just done at https://github.com/oss-review-toolkit/ort/pull/3477.

Reply:
done for `wiremock`.
For `service`, I cannot to a one liner because of dependencies construction.
Do you have a suggestion ?

Reply:
Well, it would be easier if we had a `create()` function in `FossIdRestService` to call here, just like we have such a function in `NexusIqService`, `ClearlyDefinedService` and `AmazonOssAttributionBuilderService`. Mind adding that function in a preceding commit to make use of it in this commit?

Reply:
I did the change to add a `create` function, so it looks like other ORT services.

However this still does not prevent the `lateinit` because we can construct the `service` object only after Wiremock server has started.

Reply:
> However this still does not prevent the `lateinit` because we can construct the `service` object only after Wiremock server has started.

Why? Calling `FossIdRestService.create()` doesn't actually make any use of a running WireMock server, so directly initializing the `service` class variable should be fine AFAICS.

Reply:
yes it does because it needs the dynamic port of a running Wiremock in the url

```beforeSpec` {
        wiremock.start()
        WireMock.configureFor(wiremock.port())
        service = FossIdRestService.create("http://localhost:${wiremock.port()}")
    }```

Reply:
Ah, right, I overlooked that. So let's keep this as-is.